confluent_kafka/src/confluent_kafka.c

```diff
@@ -1568,6 +1568,40 @@ rd_kafka_conf_t *common_conf_setup (rd_kafka_type_t ktype,
         /* Enable valid offsets in delivery reports */
         rd_kafka_topic_conf_set(tconf, "produce.offset.report", "true", NULL, 0);
 
+        /* Preload Plugins since dicts are unordered */
```

Comment:
Preloading a library/plugin is something else, I think you can instead flesh out the comment a bit, something like:
Plugins must be configured prior to any configuration properties they support, since dicts are unordered
we explicitly check for, set, and delete the plugin paths here to make sure they are applied prior to
their configuration.